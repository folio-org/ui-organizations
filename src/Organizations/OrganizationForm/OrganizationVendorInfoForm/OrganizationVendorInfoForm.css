.vendorInfoCheckbox {
  padding-left: 8px;
}
