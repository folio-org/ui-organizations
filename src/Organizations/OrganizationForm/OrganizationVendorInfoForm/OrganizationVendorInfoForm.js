import React, { useCallback } from 'react';
import { FormattedMessage } from 'react-intl';
import { Field, useForm } from 'react-final-form';
import { find } from 'lodash';

import {
  Checkbox,
  TextField,
  AccordionSet,
  Accordion,
  Row,
  Col,
  useCurrencyOptions,
} from '@folio/stripes/components';
import {
  FieldMultiSelectionFinal,
  FieldSelectFinal,
  PAYMENT_METHOD_OPTIONS,
} from '@folio/stripes-acq-components';

import css from './OrganizationVendorInfoForm.css';

const subSections = { taxSection: true };

const OrganizationVendorInfoForm = () => {
  const { change } = useForm();
  const currenciesOptions = useCurrencyOptions();
  const currencyValueOptions = currenciesOptions.map(({ value }) => value);
  const currencyToString = useCallback(item => item, []);
  const currencyFormatter = useCallback(({ option }) => {
    const item = find(currenciesOptions, { value: option }) || option;

    if (!item) return option;

    return item.label;
  }, [currenciesOptions]);

  const currencyFilter = useCallback((filterText) => {
    const lowerFilterText = (filterText || '').toLowerCase();
    const renderedItems = filterText
      ? currenciesOptions
        .filter(item => item.label.toLowerCase().includes(lowerFilterText))
        .map(({ value }) => value)
      : currencyValueOptions;

    return { renderedItems };
  }, [currenciesOptions, currencyValueOptions]);

  const onChangeSelect = useCallback((e) => {
    change('vendorCurrencies', e);
  }, [change]);

  return (
    <Row className={css.vendorInfo}>
      <Col xs={12}>
        <Row>
          <Col
            data-test-payment-method
            xs={6}
            md={3}
          >
            <FieldSelectFinal
              dataOptions={PAYMENT_METHOD_OPTIONS}
              label={<FormattedMessage id="ui-organizations.vendorInfo.paymentMethod" />}
              name="paymentMethod"
              validateFields={[]}
            />
          </Col>
          <Col
            data-test-currency
            xs={6}
            md={3}
          >
            <FieldMultiSelectionFinal
              ariaLabelledBy="vendorFormCurrenciesLabel"
              label={<FormattedMessage id="ui-organizations.vendorInfo.vendorCurrencies" />}
              name="vendorCurrencies"
              dataOptions={currencyValueOptions}
              formatter={currencyFormatter}
              itemToString={currencyToString}
              filter={currencyFilter}
              onChange={onChangeSelect}
              validateFields={[]}
            />
          </Col>
          <Col
            data-test-activation-interval
            xs={6}
            md={3}
          >
            <Field
              component={TextField}
              fullWidth
              id="expectedActivationInterval"
              label={<FormattedMessage id="ui-organizations.vendorInfo.expectedActivationInterval" />}
              name="expectedActivationInterval"
              type="number"
              validateFields={[]}
            />
          </Col>
          <Col
            data-test-invoice-interval
            xs={6}
            md={3}
          >
            <Field
              component={TextField}
              fullWidth
              id="expectedInvoiceInterval"
              label={<FormattedMessage id="ui-organizations.vendorInfo.expectedInvoiceInterval" />}
              name="expectedInvoiceInterval"
              type="number"
              validateFields={[]}
            />
          </Col>
          <Col
            data-test-claiming-interval
            xs={6}
            md={3}
          >
            <Field
              component={TextField}
              fullWidth
              label={<FormattedMessage id="ui-organizations.vendorInfo.claimingInterval" />}
              name="claimingInterval"
              type="number"
              validateFields={[]}
            />
          </Col>
          <Col
            data-test-receipt-interval
            xs={6}
            md={3}
          >
            <Field
              component={TextField}
              fullWidth
              id="expectedReceiptInterval"
              label={<FormattedMessage id="ui-organizations.vendorInfo.expectedReceiptInterval" />}
              name="expectedReceiptInterval"
              type="number"
              validateFields={[]}
            />
          </Col>
          <Col
            data-test-discount-percent
            xs={6}
            md={3}
          >
            <Field
              label={<FormattedMessage id="ui-organizations.vendorInfo.discountPercent" />}
              name="discountPercent"
              component={TextField}
              fullWidth
              validateFields={[]}
            />
          </Col>
          <Col
            data-test-reneval-interval
            xs={6}
            md={3}
          >
            <Field
              component={TextField}
              fullWidth
              id="renewalActivationInterval"
              label={<FormattedMessage id="ui-organizations.vendorInfo.renewalActivationInterval" />}
              name="renewalActivationInterval"
              type="number"
              validateFields={[]}
            />
          </Col>
          <Col
            data-test-subscription-interval
            xs={6}
            md={3}
          >
            <Field
              component={TextField}
              fullWidth
              id="subscriptionInterval"
              label={<FormattedMessage id="ui-organizations.vendorInfo.subscriptionInterval" />}
              name="subscriptionInterval"
              type="number"
              validateFields={[]}
            />
          </Col>
          <Col
            data-test-export-to-accounting
            xs={6}
            md={3}
          >
            <Field
              component={Checkbox}
              id="exportToAccounting"
              label={<FormattedMessage id="ui-organizations.vendorInfo.exportToAccounting" />}
              name="exportToAccounting"
              type="checkbox"
              vertical
              validateFields={[]}
            />
          </Col>
        </Row>
      </Col>

      <Col xs={12}>
        <AccordionSet accordionStatus={subSections}>
          <Accordion
            label={<FormattedMessage id="ui-organizations.vendorInfo.tax" />}
            id="taxSection"
          >
            <Row>
              <Col
                data-test-tax-id
                xs={6}
                md={3}
              >
                <Field
                  component={TextField}
                  fullWidth
                  id="taxId"
                  label={<FormattedMessage id="ui-organizations.vendorInfo.taxID" />}
                  name="taxId"
                  validateFields={[]}
                />
              </Col>
              <Col
                data-test-tax-percentage
                xs={6}
                md={3}
              >
                <Field
                  label={<FormattedMessage id="ui-organizations.vendorInfo.taxPercentage" />}
                  name="taxPercentage"
                  id="taxPercentage"
                  type="number"
                  component={TextField}
                  fullWidth
                  validateFields={[]}
                />
              </Col>
              <Col
                data-test-liable-for-vat
                xs={6}
                md={3}
              >
                <Field
                  label={<FormattedMessage id="ui-organizations.vendorInfo.liableForVAT" />}
                  name="liableForVat"
                  id="liableForVat"
                  component={Checkbox}
                  type="checkbox"
                  vertical
                  validateFields={[]}
                />
              </Col>
            </Row>
          </Accordion>
        </AccordionSet>
      </Col>
    </Row>
  );
};

export default OrganizationVendorInfoForm;
