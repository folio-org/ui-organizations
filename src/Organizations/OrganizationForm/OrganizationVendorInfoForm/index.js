export { default as OrganizationVendorInfoForm } from './OrganizationVendorInfoForm';
