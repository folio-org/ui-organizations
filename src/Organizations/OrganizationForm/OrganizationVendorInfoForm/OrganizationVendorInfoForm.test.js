import React from 'react';
import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

import stripesFinalForm from '@folio/stripes/final-form';

import { organization } from '../../../../test/jest/fixtures';

import OrganizationVendorInfoForm from './OrganizationVendorInfoForm';

const TestForm = stripesFinalForm({})(
  () => {
    return (
      <form>
        <OrganizationVendorInfoForm />
      </form>
    );
  },
);

const renderForm = ({ initialValues = {} } = {}) => render(
  <TestForm
    onSubmit={jest.fn()}
    initialValues={initialValues}
  />,
  { wrapper: MemoryRouter },
);

describe('OrganizationVendorInfoForm', () => {
  beforeEach(() => {
    global.document.createRange = global.document.originalCreateRange;
  });

  afterEach(() => {
    global.document.createRange = global.document.mockCreateRange;
  });

  it('should render correct structure', async () => {
    const { asFragment } = renderForm({ initialValues: organization });

    await screen.findByText('ui-organizations.vendorInfo.paymentMethod');

    expect(asFragment()).toMatchSnapshot();
  });
});
