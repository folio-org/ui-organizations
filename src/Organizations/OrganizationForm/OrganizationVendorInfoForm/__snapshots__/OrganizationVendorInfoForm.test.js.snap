// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrganizationVendorInfoForm should render correct structure 1`] = `
<DocumentFragment>
  <form>
    <div
      class="vendorInfo row"
    >
      <div
        class="col-xs-12"
      >
        <div
          class="row"
        >
          <div
            class="col-xs-6 col-md-3"
            data-test-payment-method="true"
          >
            <div
              class="select inputGroup fullWidth"
            >
              <label
                class="label"
                for="select-1"
                id="select-1-label"
              >
                ui-organizations.vendorInfo.paymentMethod
              </label>
              <div
                class="selectWrap"
              >
                <select
                  aria-invalid="false"
                  aria-labelledby="select-1-label"
                  aria-required="false"
                  class="selectControl formControl"
                  id="select-1"
                  name="paymentMethod"
                >
                  <option
                    value=""
                  >
                    stripes-acq-components.label.emptyValue
                  </option>
                  <option
                    value="Cash"
                  >
                    stripes-acq-components.paymentMethod.cash
                  </option>
                  <option
                    value="Credit Card"
                  >
                    stripes-acq-components.paymentMethod.card
                  </option>
                  <option
                    value="EFT"
                  >
                    stripes-acq-components.paymentMethod.eft
                  </option>
                  <option
                    value="Deposit Account"
                  >
                    stripes-acq-components.paymentMethod.depAccount
                  </option>
                  <option
                    value="Physical Check"
                  >
                    stripes-acq-components.paymentMethod.physicalCheck
                  </option>
                  <option
                    value="Bank Draft"
                  >
                    stripes-acq-components.paymentMethod.bankDraft
                  </option>
                  <option
                    value="Internal Transfer"
                  >
                    stripes-acq-components.paymentMethod.internalTransfer
                  </option>
                  <option
                    value="Other"
                  >
                    stripes-acq-components.paymentMethod.other
                  </option>
                </select>
                <span>
                  Icon
                </span>
              </div>
              <div
                role="alert"
              />
            </div>
          </div>
          <div
            class="col-xs-6 col-md-3"
            data-test-currency="true"
          >
            <div
              class="multiSelectContainer"
              id="multiselect-1"
            >
              <div
                aria-live="assertive"
                aria-relevant="additions"
                class="sr-only"
              />
              <div
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-labelledby="vendorFormCurrenciesLabel"
                class="multiSelectContainer"
                role="application"
              >
                <label
                  class="label"
                  for="multiselect-1-input"
                  id="vendorFormCurrenciesLabel"
                >
                  ui-organizations.vendorInfo.vendorCurrencies
                </label>
                <span
                  class="sr-only"
                  id="multi-value-status-multiselect-1"
                >
                  1 item selected
                </span>
                <span
                  class="sr-only"
                  id="multi-describe-control-multiselect-1"
                >
                  stripes-components.multiSelection.controlDescription
                </span>
                <div
                  class="multiSelectControlWrapper"
                >
                  <div
                    aria-describedby="multi-describe-control-multiselect-1"
                    aria-labelledby="vendorFormCurrenciesLabel multi-value-status-multiselect-1"
                    class="multiSelectControl formControl"
                    tabindex="0"
                  >
                    <div
                      class="multiSelectControlGroup"
                    >
                      <div
                        class="multiSelectValueListContainer"
                        id="multi-values-list-multiselect-1"
                      >
                        <span
                          class="sr-only"
                          id="multi-value-multiselect-1"
                        >
                          stripes-components.multiSelection.removeSelectedButtonLabel
                        </span>
                        <span
                          class="sr-only"
                          id="multi-describe-action-multiselect-1"
                        >
                          stripes-components.multiSelection.removeSelectedButtonDescription
                        </span>
                        <ul
                          class="multiSelectValueList"
                        >
                          <li
                            class="valueChipRoot"
                            id="USD-0_multiselect_selected"
                          >
                            <div
                              class="valueChipGroup"
                            >
                              <div
                                class="valueChipValue"
                                id="USD-0_multiselect_selected_label"
                              >
                                USD (USD)
                              </div>
                              <button
                                aria-describedby="multi-describe-action-multiselect-1"
                                aria-label="times"
                                aria-labelledby="USD-0_multiselect_selected_label multi-value-multiselect-1"
                                class="iconButton small"
                                icon="times"
                                style="padding: 0px;"
                                type="button"
                              >
                                <span
                                  aria-describedby="multi-describe-action-multiselect-1"
                                  class="iconButtonInner smallOffset"
                                >
                                  <span>
                                    Icon
                                  </span>
                                </span>
                              </button>
                            </div>
                          </li>
                        </ul>
                      </div>
                      <input
                        aria-autocomplete="list"
                        aria-labelledby="vendorFormCurrenciesLabel"
                        autocomplete="off"
                        class="multiSelectFilterField"
                        id="multiselect-1-input"
                        placeholder=""
                        type="text"
                        value=""
                      />
                      <input
                        aria-autocomplete="list"
                        aria-hidden="true"
                        autocomplete="off"
                        class="multiSelectValueInput"
                        id="multiselect-input-multiselect-1"
                        tabindex="-1"
                        type="text"
                        value="USD"
                      />
                    </div>
                    <button
                      aria-haspopup="true"
                      aria-label="open menu"
                      class="multiSelectToggleButton"
                      data-toggle="true"
                      role="button"
                      type="button"
                    >
                      <div
                        class="textFieldIcon"
                      >
                        <span>
                          Icon
                        </span>
                      </div>
                    </button>
                  </div>
                  <div
                    class="multiSelectMenu"
                    hidden=""
                    style="width: 100px;"
                  >
                    <div
                      class="multiSelectFeedback"
                      role="alert"
                    />
                    <ul
                      aria-labelledby="vendorFormCurrenciesLabel"
                      class="multiSelectOptionList"
                      id="multiselect-option-list-multiselect-1"
                      role="listbox"
                      style="max-height: 168px;"
                    >
                      <li
                        aria-selected="false"
                        class="multiSelectOption optionCursor"
                        id="multiselect-1-main-item-0"
                        role="option"
                      >
                        BYN (BYN)
                        <div>
                          +
                        </div>
                      </li>
                      <li
                        aria-selected="true"
                        class="multiSelectOption optionSelected"
                        id="multiselect-1-main-item-1"
                        role="option"
                      >
                        USD (USD)
                        <div>
                          -
                        </div>
                      </li>
                    </ul>
                  </div>
                </div>
              </div>
              <div
                role="alert"
              />
            </div>
          </div>
          <div
            class="col-xs-6 col-md-3"
            data-test-activation-interval="true"
          >
            <div
              class="textField"
            >
              <label
                class="label"
                for="expectedActivationInterval"
              >
                ui-organizations.vendorInfo.expectedActivationInterval
              </label>
              <div
                class="formControl inputGroup"
              >
                <input
                  aria-invalid="false"
                  autocapitalize="off"
                  autocomplete="off"
                  autocorrect="off"
                  error=""
                  id="expectedActivationInterval"
                  name="expectedActivationInterval"
                  spellcheck="false"
                  type="number"
                  value="0"
                  warning=""
                />
              </div>
              <div
                role="alert"
              />
            </div>
          </div>
          <div
            class="col-xs-6 col-md-3"
            data-test-invoice-interval="true"
          >
            <div
              class="textField"
            >
              <label
                class="label"
                for="expectedInvoiceInterval"
              >
                ui-organizations.vendorInfo.expectedInvoiceInterval
              </label>
              <div
                class="formControl inputGroup"
              >
                <input
                  aria-invalid="false"
                  autocapitalize="off"
                  autocomplete="off"
                  autocorrect="off"
                  error=""
                  id="expectedInvoiceInterval"
                  name="expectedInvoiceInterval"
                  spellcheck="false"
                  type="number"
                  value="0"
                  warning=""
                />
              </div>
              <div
                role="alert"
              />
            </div>
          </div>
          <div
            class="col-xs-6 col-md-3"
            data-test-claiming-interval="true"
          >
            <div
              class="textField"
            >
              <label
                class="label"
                for="text-input-2"
              >
                ui-organizations.vendorInfo.claimingInterval
              </label>
              <div
                class="formControl inputGroup"
              >
                <input
                  aria-invalid="false"
                  autocapitalize="off"
                  autocomplete="off"
                  autocorrect="off"
                  error=""
                  id="text-input-2"
                  name="claimingInterval"
                  spellcheck="false"
                  type="number"
                  value="45"
                  warning=""
                />
              </div>
              <div
                role="alert"
              />
            </div>
          </div>
          <div
            class="col-xs-6 col-md-3"
            data-test-receipt-interval="true"
          >
            <div
              class="textField"
            >
              <label
                class="label"
                for="expectedReceiptInterval"
              >
                ui-organizations.vendorInfo.expectedReceiptInterval
              </label>
              <div
                class="formControl inputGroup"
              >
                <input
                  aria-invalid="false"
                  autocapitalize="off"
                  autocomplete="off"
                  autocorrect="off"
                  error=""
                  id="expectedReceiptInterval"
                  name="expectedReceiptInterval"
                  spellcheck="false"
                  type="number"
                  value="30"
                  warning=""
                />
              </div>
              <div
                role="alert"
              />
            </div>
          </div>
          <div
            class="col-xs-6 col-md-3"
            data-test-discount-percent="true"
          >
            <div
              class="textField"
            >
              <label
                class="label"
                for="text-input-3"
              >
                ui-organizations.vendorInfo.discountPercent
              </label>
              <div
                class="formControl inputGroup"
              >
                <input
                  aria-invalid="false"
                  autocapitalize="off"
                  autocomplete="off"
                  autocorrect="off"
                  error=""
                  id="text-input-3"
                  name="discountPercent"
                  spellcheck="false"
                  type="text"
                  value="0"
                  warning=""
                />
              </div>
              <div
                role="alert"
              />
            </div>
          </div>
          <div
            class="col-xs-6 col-md-3"
            data-test-reneval-interval="true"
          >
            <div
              class="textField"
            >
              <label
                class="label"
                for="renewalActivationInterval"
              >
                ui-organizations.vendorInfo.renewalActivationInterval
              </label>
              <div
                class="formControl inputGroup"
              >
                <input
                  aria-invalid="false"
                  autocapitalize="off"
                  autocomplete="off"
                  autocorrect="off"
                  error=""
                  id="renewalActivationInterval"
                  name="renewalActivationInterval"
                  spellcheck="false"
                  type="number"
                  value="0"
                  warning=""
                />
              </div>
              <div
                role="alert"
              />
            </div>
          </div>
          <div
            class="col-xs-6 col-md-3"
            data-test-subscription-interval="true"
          >
            <div
              class="textField"
            >
              <label
                class="label"
                for="subscriptionInterval"
              >
                ui-organizations.vendorInfo.subscriptionInterval
              </label>
              <div
                class="formControl inputGroup"
              >
                <input
                  aria-invalid="false"
                  autocapitalize="off"
                  autocomplete="off"
                  autocorrect="off"
                  error=""
                  id="subscriptionInterval"
                  name="subscriptionInterval"
                  spellcheck="false"
                  type="number"
                  value="0"
                  warning=""
                />
              </div>
              <div
                role="alert"
              />
            </div>
          </div>
          <div
            class="col-xs-6 col-md-3"
            data-test-export-to-accounting="true"
          >
            <div
              class="checkbox inline vertical"
              data-test-checkbox="true"
            >
              <label
                class="label label checkboxInteractionStylesControl"
                for="exportToAccounting"
              >
                <span
                  class="labelText"
                >
                  ui-organizations.vendorInfo.exportToAccounting
                </span>
                <span
                  class="inner checkboxInteractionStyles"
                >
                  <input
                    aria-invalid="false"
                    checked=""
                    class="input"
                    id="exportToAccounting"
                    name="exportToAccounting"
                    type="checkbox"
                    value=""
                  />
                  <span
                    class="checkboxIcon"
                  >
                    <svg
                      height="10"
                      viewBox="0 0 14 14"
                      width="10"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M10.395 4.617L5.63 9.383 3.605 7.358"
                        fill="none"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-miterlimit="10"
                        stroke-width="1.5"
                      />
                    </svg>
                  </span>
                </span>
              </label>
              <div
                role="alert"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="col-xs-12"
      >
        <div
          aria-multiselectable="true"
        >
          <section
            class="accordion hasSeparator"
            data-test-accordion-section="true"
            id="taxSection"
          >
            <div
              class="headerWrapper"
              tabindex="-1"
            >
              <div
                class="header default"
              >
                <h3
                  class="headline size-medium margin-none font-weight-bold block"
                  data-test-headline="true"
                >
                  <button
                    aria-controls="accordion2"
                    aria-expanded="true"
                    class="defaultCollapseButton"
                    id="accordion-toggle-button-taxSection"
                    type="button"
                  >
                    <span
                      class="headerInner"
                    >
                      <span
                        class="defaultHeaderIcon"
                      >
                        <span>
                          Icon
                        </span>
                      </span>
                      <div
                        class="labelArea"
                      >
                        ui-organizations.vendorInfo.tax
                      </div>
                    </span>
                  </button>
                </h3>
              </div>
            </div>
            <div
              class="content-wrap expanded"
              style="z-index: 1;"
            >
              <div
                aria-labelledby="accordion-toggle-button-taxSection"
                class="content-region expanded"
                data-test-accordion-wrapper="true"
                id="accordion2"
                role="region"
              >
                <div
                  class="row"
                >
                  <div
                    class="col-xs-6 col-md-3"
                    data-test-tax-id="true"
                  >
                    <div
                      class="textField"
                    >
                      <label
                        class="label"
                        for="taxId"
                      >
                        ui-organizations.vendorInfo.taxID
                      </label>
                      <div
                        class="formControl inputGroup"
                      >
                        <input
                          aria-invalid="false"
                          autocapitalize="off"
                          autocomplete="off"
                          autocorrect="off"
                          error=""
                          id="taxId"
                          name="taxId"
                          spellcheck="false"
                          type="text"
                          value=""
                          warning=""
                        />
                      </div>
                      <div
                        role="alert"
                      />
                    </div>
                  </div>
                  <div
                    class="col-xs-6 col-md-3"
                    data-test-tax-percentage="true"
                  >
                    <div
                      class="textField"
                    >
                      <label
                        class="label"
                        for="taxPercentage"
                      >
                        ui-organizations.vendorInfo.taxPercentage
                      </label>
                      <div
                        class="formControl inputGroup"
                      >
                        <input
                          aria-invalid="false"
                          autocapitalize="off"
                          autocomplete="off"
                          autocorrect="off"
                          error=""
                          id="taxPercentage"
                          name="taxPercentage"
                          spellcheck="false"
                          type="number"
                          value="0"
                          warning=""
                        />
                      </div>
                      <div
                        role="alert"
                      />
                    </div>
                  </div>
                  <div
                    class="col-xs-6 col-md-3"
                    data-test-liable-for-vat="true"
                  >
                    <div
                      class="checkbox inline vertical"
                      data-test-checkbox="true"
                    >
                      <label
                        class="label label checkboxInteractionStylesControl"
                        for="liableForVat"
                      >
                        <span
                          class="labelText"
                        >
                          ui-organizations.vendorInfo.liableForVAT
                        </span>
                        <span
                          class="inner checkboxInteractionStyles"
                        >
                          <input
                            aria-invalid="false"
                            class="input"
                            id="liableForVat"
                            name="liableForVat"
                            type="checkbox"
                            value=""
                          />
                          <span
                            class="checkboxIcon"
                          >
                            <svg
                              height="10"
                              viewBox="0 0 14 14"
                              width="10"
                              xmlns="http://www.w3.org/2000/svg"
                            >
                              <path
                                d="M10.395 4.617L5.63 9.383 3.605 7.358"
                                fill="none"
                                stroke-linecap="round"
                                stroke-linejoin="round"
                                stroke-miterlimit="10"
                                stroke-width="1.5"
                              />
                            </svg>
                          </span>
                        </span>
                      </label>
                      <div
                        role="alert"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </section>
        </div>
      </div>
    </div>
  </form>
  <div />
</DocumentFragment>
`;
